lib/wanda/catalog.ex

```diff
@@ -22,26 +24,37 @@ defmodule Wanda.Catalog do
     |> Path.join("/*")
     |> Path.wildcard()
     |> Enum.map(&Path.basename(&1, ".yaml"))
-    |> Enum.map(&get_check(&1))
+    |> get_checks()
   end
 
   @doc """
   Get a check from the catalog.
   """
-  @spec get_check(String.t()) :: Check.t()
+  @spec get_check(String.t()) :: {:ok, Check.t()} | {:error, any}
   def get_check(check_id) do
-    get_catalog_path()
-    |> Path.join("#{check_id}.yaml")
-    |> YamlElixir.read_from_file!()
-    |> map_check()
+    case get_catalog_path()
+         |> Path.join("#{check_id}.yaml")
+         |> YamlElixir.read_from_file() do
+      {:ok, file_content} ->
+        {:ok, map_check(file_content)}
```

Comment:
Now I'm wondering if we should do some sanitiy check with `map_check`. I mean, this function will fail if it doesn't receive some expected fields (like `name` missing in check).
Maybe we should be resilient as well here.

It would be as simple as adding other case as `map_check(_)` which returns a `{:error, :malformed_check}` or similar, and tune `map_check` to return `{:ok, check}` otherwise.